File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/maintenance/ReportTheftUseCase.java
package com.lattis.ellipse.domain.interactor.maintenance;

import com.lattis.ellipse.data.network.model.response.BasicResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.MaintenanceRepository;

import javax.inject.Inject;

import io.reactivex.Observable;



public class ReportTheftUseCase extends UseCase<BasicResponse> {
    private int bikeId;
    private int tripId=-1;
    private MaintenanceRepository maintenanceRepository;
    @Inject
    protected ReportTheftUseCase(ThreadExecutor threadExecutor,
                                PostExecutionThread postExecutionThread,
                                MaintenanceRepository maintenanceRepository) {
        super(threadExecutor, postExecutionThread);
        this.maintenanceRepository = maintenanceRepository;
    }

    @Override
    protected Observable<BasicResponse> buildUseCaseObservable() {
        return maintenanceRepository.reportBikeTheft(bikeId,tripId);

    }

    public ReportTheftUseCase withBikeId(int bikeId) {
        this.bikeId = bikeId;
        return this;
    }

    public ReportTheftUseCase withTripId(int tripId) {
        this.tripId = tripId;
        return this;
    }

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/maintenance/ReportTheftUseCase.kt
package com.lattis.domain.usecase.maintenance

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.MaintenanceRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class ReportTheftUseCase @Inject protected constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val maintenanceRepository: MaintenanceRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {
    private var bikeId = 0
    private var tripId = -1
    override fun buildUseCaseObservable(): Observable<Boolean> {
        return maintenanceRepository.reportTheft(bikeId, tripId)
    }

    fun withBikeId(bikeId: Int): ReportTheftUseCase {
        this.bikeId = bikeId
        return this
    }

    fun withTripId(tripId: Int): ReportTheftUseCase {
        this.tripId = tripId
        return this
    }

}